### PR TITLE
Add slider to adjust capacity_adjustment

### PR DESCRIPTION
**Pull request by nixocio**
Add slider to adjust capacity adjustment.

Also add a new custom hook, useDebounce.

image

See: https://github.com/ansible/awx/issues/7777

### CODE REVIEW

File: awx/ui_next/src/screens/InstanceGroup/Instances/InstanceListItem.jsx [hunk 1]
```diff
@@ -56,81 +103,145 @@ function InstanceListItem({
...
+                aria-label={t`capacity adjustment`}
+                width={4}
+              >
+                <SliderHolder data-cy="slider-holder">
+                  <div data-cy="cpu-capacity">{t`CPU ${instance.cpu_capacity}`}</div>
+                  <SliderForks data-cy="slider-forks">
+                    <div data-cy="number-forks">
+                      <Plural value={forks} one="# fork" other="# forks" />
+                    </div>
+                    <Slider
```
Comment (nixocio) [inline awx/ui_next/src/screens/InstanceGroup/Instances/InstanceListItem.jsx:189]: Slider does not support ouiaId yet. @tiagodread I added `data-cy` in order to ease testing. Let me know whether they are good enough.

**Comment by nixocio** [comment]
Running e2e tests.

File: awx/ui_next/CONTRIBUTING.md [hunk 2]
```diff
@@ -352,7 +352,7 @@ You can learn more about the ways lingui and its React helpers at [this link](ht
 ### Setting up .po files to give to translation team
 
 1. `npm run add-locale` to add the language that you want to translate to (we should only have to do this once and the commit to repo afaik). Example: `npm run add-locale en es fr` # Add English, Spanish and French locale
-2. `npm run extract-strings` to create .po files for each language specified. The .po files will be placed in src/locales. When updating strings that are used by `<Plural>` or `plural()` you will need to run this command to get the strings to render properly. This commmand will create `.po` files for each of the supported languages that will need to be commited with your PR.
+2. `npm run extract-strings` to create .po files for each language specified. The .po files will be placed in src/locales. When updating strings that are used by `<Plural>` or `plural()` you will need to run this command to get the strings to render properly. This command will create `.po` files for each of the supported languages that will need to be committed with your PR.
```
Comment (AlexSCorey) [inline awx/ui_next/CONTRIBUTING.md:353]: Good catch

File: awx/ui_next/src/screens/InstanceGroup/Instances/InstanceListItem.test.jsx [hunk 3]
```diff
@@ -68,6 +109,19 @@ describe('<InstanceListItem/>', () => {
       wrapper.find('PFDataListCell[aria-label="instance type"]').text()
     ).toBe('TypeAuto');
     expect(wrapper.find('input#instances-1').prop('checked')).toBe(false);
+    expect(
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/InstanceGroup/Instances/InstanceListItem.test.jsx:124]: Can we add a test for error handling?

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
Screen Shot 2021-05-03 at 12 52 09 PM
Can we get them items in blue boxes, as well as the CPU and RAM values to align with each other.  Can we get the items in the green boxes to align with each other?

Screen Shot 2021-05-03 at 1 09 46 PM

**Comment by nixocio** [comment]
> Screen Shot 2021-05-03 at 12 52 09 PM
> Can we get them items in blue boxes, as well as the CPU and RAM values to align with each other. Can we get the items in the green boxes to align with each other?
> 
> Screen Shot 2021-05-03 at 1 09 46 PM

I followed with @trahman73 about the alignment used on the mockups,  his feedback - All components should be vertically aligned on the middle. Updated accordingly.

image

File: awx/ui_next/src/util/useDebounce.test.jsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+function Test({ fn, delay = 500, data }) {
+  const debounce = useDebounce(fn, delay);
+  debounce(data);
+  return <TestInner />;
+}
+
+test('useDebounce', () => {
+  jest.useFakeTimers();
+  const fn = jest.fn();
+  expect(fn).toHaveBeenCalledTimes(0);
```
Comment (keithjgrant) [inline awx/ui_next/src/util/useDebounce.test.jsx]: you should put this first assertion after `mount`

Comment (nixocio) [reply awx/ui_next/src/util/useDebounce.test.jsx]: @keithjgrant updated.

**Review by AlexSCorey — APPROVED** [review]
Looks good to me.

**Review by keithjgrant — APPROVED** [review]

**Review by tiagodread — APPROVED** [review]
I've verified manually, we are following the same behaviour like the old UI, I'm still working on e2e test for this change.

**Comment by tiagodread** [comment]
@nixocio we can merge this PR after a rebase

**Pull request merged**